pkg/resource/deserializer.go

```diff
@@ -0,0 +1,24 @@
+package resource
+
+import (
+	"github.com/zclconf/go-cty/cty"
+)
+
+type Deserializer struct {
+	factory ResourceFactory
+}
+
+func NewDeserializer(factory ResourceFactory) *Deserializer {
+	return &Deserializer{factory}
+}
+
+func (s *Deserializer) Deserialize(ty string, rawList []cty.Value) ([]Resource, error) {
+	resources := make([]Resource, 0)
+	for _, rawResource := range rawList {
+		rawResource := rawResource
+		attrs := ToResourceAttributes(&rawResource)
```

Comment:
What's the purpose of `ToResourceAttributes` ? This logic do not need to be in a dedicated exported method, it does a deserialization of cty values object and it is exactly the purpose of this deserializer.

Reply:
yeah you're right. I think this is a legacy method that we did not move